lib/validator/formchecker.cpp

```diff
@@ -229,16 +229,23 @@ Expect<void> FormChecker::checkInstr(const AST::Instruction &Instr) {
   // Helper lambda for checking memory alignment and perform transformation.
   auto checkAlignAndTrans = [this, checkLaneAndTrans,
                              &Instr](uint32_t N, Span<const ValType> Take,
-                                     Span<const ValType> Put,
+                                     Span<const ValType> Put, bool IsAtomic,
```

Comment:
I think it's better not to use this `IsAtomic` --
Due to this lambda contains the `Instr` reference, you can check the OpCode in the lambda body.
Thus, the `IsAtomic` can be known by a simple condition:

```
if (Code >= OpCode::I32__atomic__load &&
    Code <= OpCode::I64__atomic__rmw32__cmpxchg_u) {
 ...
}
```